lib/spack/spack/solver/asp.py

```diff
@@ -3999,6 +4007,31 @@ def _specs_from_mirror():
         return []
 
 
+def _specs_from_environment(path):
+    active_env = ev.active_environment()
+
+    if path:
+        if not ev.is_env_dir(path) and ev.exists(path):
+            path = ev.root(path)
+
+    if active_env and path in active_env.included_specs_by_hash:
+
+        def _specs_for_env():
+            for s in active_env.included_specs_by_hash[path].values():
+                yield s
+
+        return _specs_for_env()
+    elif path:
+        env = ev.read(path)
+    else:
+        env = active_env
+
+    if env:
+        return map(lambda x: x[1], env.concretized_specs())
```

Comment:
```suggestion
        return [concrete for _, concrete in env.concretized_specs()]
```
I think this avoids unnecessary function call overhead and is easier to read.

Reply:
I thought that map, which in my understanding is just a generator function, would be faster since I wouldn't need to allocate the memory for all of the concretized specs. I originally had it the other way.

I agree that is is a bit more readable as a list.

Reply:
I generally prefer lists because generators are easy to write bugs with, but if you want a generator you can still use this syntax.

`return (concrete for _, concrete in env.concretized_specs())`